dashboard/app/models/script.rb

```diff
@@ -888,66 +888,80 @@ def has_older_version_progress?(user)
-        scripts_to_add << [{
-          id: script_data[:id],
-          name: name,
-          hidden: script_data[:hidden].nil? ? true : script_data[:hidden], # default true
-          login_required: script_data[:login_required].nil? ? false : script_data[:login_required], # default false
-          wrapup_video: script_data[:wrapup_video],
-          new_name: script_data[:new_name],
-          family_name: script_data[:family_name],
-          properties: Script.build_property_hash(script_data).merge(new_properties)
-        }, lesson_groups]
-      end
+    scripts_to_add = []
+
+    custom_i18n = {}
+    # Load custom scripts from Script DSL format
+    custom_files.map do |script|
+      name = File.basename(script, '.script')
+      base_name = Script.base_name(name)
+      name = "#{base_name}-#{new_suffix}" if new_suffix
+      script_data, i18n =
+        begin
+          ScriptDSL.parse_file(script, name)
+        rescue => e
+          raise e, "Error parsing script file #{script}: #{e}"
+        end
+
+      lesson_groups = script_data[:lesson_groups]
+      custom_i18n.deep_merge!(i18n)
+      # TODO: below is duplicated in update_text. and maybe can be refactored to pass script_data?
+      scripts_to_add << [{
+        id: script_data[:id],
+        name: name,
+        hidden: script_data[:hidden].nil? ? true : script_data[:hidden], # default true
+        login_required: script_data[:login_required].nil? ? false : script_data[:login_required], # default false
+        wrapup_video: script_data[:wrapup_video],
+        new_name: script_data[:new_name],
+        family_name: script_data[:family_name],
+        properties: Script.build_property_hash(script_data).merge(new_properties)
+      }, lesson_groups]
+    end
 
-      # Stable sort by ID then add each script, ensuring scripts with no ID end up at the end
-      added_scripts = scripts_to_add.sort_by.with_index {|args, idx| [args[0][:id] || Float::INFINITY, idx]}.map do |options, raw_lesson_groups|
-        add_script(options, raw_lesson_groups, new_suffix: new_suffix, editor_experiment: new_properties[:editor_experiment])
-      rescue => e
-        raise e, "Error adding script named '#{options[:name]}': #{e}", e.backtrace
+    # Print progress for dev environments and adhocs
+    debug = [:adhoc, :development].include?(rack_env)
+    if debug
+      puts "Seeding #{scripts_to_add.length} Scripts"
+      print "Seeded... "
+      n = 0
+    end
+
+    # Stable sort by ID then add each script, ensuring scripts with no ID end up at the end
+    added_script_names = scripts_to_add.sort_by.with_index {|args, idx| [args[0][:id] || Float::INFINITY, idx]}.map do |options, raw_lesson_groups|
+      added_script = add_script(options, raw_lesson_groups, new_suffix: new_suffix, editor_experiment: new_properties[:editor_experiment])
+      if debug
+        n += 1
+        print "#{n}, "
```

Comment:
I think this could be cleaner if we used one of the two progressbar gems already in our Ruby dependencies:

- [`ruby-progressbar`](https://github.com/jfelchner/ruby-progressbar) (only in `group :development` in `Gemfile` right now, we'd need to add it to all environments): 
```ruby
progress = ProgressBar.create(total: scripts_to_add.length)
# in loop
progress.increment
```
- [`progress`](https://github.com/toy/progress) (implicit dependency of `image_optim`, we could make it explicit):
```ruby
added_script_names = scripts_to_add.sort_by.with_index {[...]}.with_progress.map do
```

Reply:
Nice! updated.